packages/security/src/K8sCredentialManager.ts

```diff
@@ -0,0 +1,229 @@
+            await this.kc.createNamespacedSecret(this.kubeConfig.namespace, {
+                "apiVersion": "v1",
+                "kind": "Secret",
+                "metadata": {
+                    "name": `${this.secretName}`,
+                    "namespace": `${this.kubeConfig.namespace}`
+                },
+                "type": "Opaque",
+                "data": {
+                    "credentials": credentials
+                }
+            }, "true");
+            Logger.getImperativeLogger().debug(`Successfully stored credentials as a kubernetes secret on namespace ${this.kubeConfig.namespace}`);
+        } catch (err) {
+            throw new ImperativeError({
+                msg: `Error when saving k8s secret ${this.secretName}`,
+                additionalDetails: err
+            });
+        }
+    }
+
+    /**
+     * Calls the K8s API to delete secrets with {@link K8sCredentialManager#service} and the
+     * account is passed to the function by Imperative.
+     *
+     * @param {string} account The account to set credentials
+     *
+     * @returns {Promise<void>} A promise that the function has completed.
+     *
+     * @throws {@link ImperativeError} if call to K8s API fails.
+     */
+    protected async deleteCredentials(account: string): Promise<void> {
+        try {
+            await this.readNamespacedSecret();
+            Logger.getImperativeLogger().debug(`Deleting k8s secret ${this.secretName}`);
+            await this.kc.deleteNamespacedSecret(this.secretName, this.kubeConfig.namespace, "true");
+        } catch (err) {
+            throw new ImperativeError({
+                msg: `Failed to delete secret ${this.secretName} in namespace ${this.kubeConfig.namespace}`,
+                additionalDetails: err.message
+            });
+        }
+    }
+
+    private get defaultService(): string {
+        return this.allServices[0];
+    }
+
+    private getMissingEntryMessage(account: string) {
+        let listOfServices = `  Service = `;
+        for (const service of this.allServices) {
+            listOfServices += `${service}, `;
+        }
+        const commaAndSpace = 2;
+        listOfServices = listOfServices.slice(0, -1 * commaAndSpace) + `\n  Account = ${account}\n\n`;
+
+        return "Could not find an entry in the credential vault for the following:\n" +
+            listOfServices +
+            "Possible Causes:\n" +
+            "  This could have been caused by any manual removal of credentials from your vault.\n\n" +
+            "Resolutions: \n" +
+            "  Recreate the credentials in the vault for the particular service in the vault.\n";
+    }
+
+    /**
+     * Retrieve data from kube config and return a {@link KubeConfig} object.
+     *
+     * @returns {k8s.CoreV1Api} the object allowing access to k8s api calls
+     *
+     * @throws {@link ImperativeError} if kube config file was not able to be opened.
+     */
+    private setupKubeConfig(): any {
```

Comment:
Should we provide a means for users to define a hostname for their K8s cluster, or is it safe to assume for now that the cluster exists on the same machine where Zowe CLI is being run? I'm not a K8s expert, so please ignore this comment if it doesn't make sense 😛

Reply:
The hostname does not need to be defined since the hostname is pulled from the current session when logging into the desired k8s cluster, if what you meant is the namespace, you can switch contexts in with a `kubectl config set-context --current --namespace="my-namespace-name"` and this will be the location where secrets will be stored/retrieved from without having to define it from the Zowe side.